### PR TITLE
feat(plugin-js-packages): set up and implement configuration for audit plugin

### PR DESCRIPTION
**Pull request by Tlacenka**
Related to #542 

In this PR I do the following:
- Set up the plugin, its ESLint, dependencies, etc.
- Define plugin configuration schema.
- Create documentation.

Notes:
- Runner will follow in another PR.
- For now, all dependency groups are checked. Configurable dependency groups will be in post-MVP.
- Naming improvements are welcome!

### CODE REVIEW

File: packages/plugin-js-packages/src/lib/config.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// ): Record<PackageAuditLevel, IssueSeverity> {
+//   return Object.fromEntries(
+//     objectKeys(defaultAuditLevelMapping).map<
+//       [PackageAuditLevel, IssueSeverity]
+//     >(auditLevel => [
+//       auditLevel,
+//       mapping[auditLevel] ?? defaultAuditLevelMapping[auditLevel],
+//     ]),
+//   );
+// }
```
Comment (matejchalk) [inline packages/plugin-js-packages/src/lib/config.ts:55]: :raised_eyebrow: :question:

File: packages/plugin-js-packages/src/lib/utils.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { PackageManager } from './config';
+
+export const pkgManagerDocs: Record<PackageManager, string> = {
+  npm: 'https://docs.npmjs.com/',
+  yarn: 'https://classic.yarnpkg.com/lang/en/docs/',
+  'yarn-berry': 'https://yarnpkg.com/getting-started',
+  pnpm: 'https://pnpm.io/pnpm-cli',
+};
+export const auditDocs: Record<PackageManager, string> = {
+  npm: 'https://docs.npmjs.com/cli/v10/commands/npm-audit',
```
Comment (matejchalk) [inline packages/plugin-js-packages/src/lib/utils.ts:10]:
```suggestion
  npm: 'https://docs.npmjs.com/cli/commands/npm-audit',
```

**Review by matejchalk — COMMENTED** [review]
Have some suggestions for what groups and audits to create. But the project setup is good, looks like you thought of everything :slightly_smiling_face:

File: packages/plugin-js-packages/README.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+📦 **Code PushUp plugin for JavaScript packages.** 🛡️
+
+This plugin allows you to list outdated dependencies and run audit for known vulnerabilities.
+It supports the following package managers: npm, yarn, yarn berry, pnpm.
+
+## Getting started
+
+1. If you haven't already, install [@code-pushup/cli](../cli/README.md) and create a configuration file.
+
+2. Insert plugin configuration. By default, npm audit and npm outdated commands will be run.
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply packages/plugin-js-packages/README.md:16]: This does not take into consideration that I put NPM as default. WDYT, should it be a default or should I make package manager required?

File: packages/plugin-js-packages/src/lib/config.ts (hunk 1 above)
Comment (Tlacenka) [reply packages/plugin-js-packages/src/lib/config.ts:55]: I attempted to fill the record dynamically and failed. But unless the audit levels change a lot, even the explicit mapping is fine. WDYT?

File: packages/plugin-js-packages/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (matejchalk) [reply packages/plugin-js-packages/README.md:16]: Good question :thinking: Maybe best to make the package manager required.

File: packages/plugin-js-packages/src/lib/config.ts (hunk 1 above)
Comment (matejchalk) [reply packages/plugin-js-packages/src/lib/config.ts:55]: The audit levels change hardly ever, I believe.

File: packages/plugin-js-packages/README.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+             plugin: 'js-packages',
+             slug: 'npm-audit', // replace prefix with your package manager
+             weight: 2,
+           },
+           {
+             type: 'audit',
+             plugin: 'js-packages',
+             slug: 'npm-outdated', // replace prefix with your package manager
+             weight: 1,
+           },
```
[the remark this answers is not in the document]
Comment (Tlacenka) [reply packages/plugin-js-packages/README.md]: Done. For now, all groups are included.

File: packages/plugin-js-packages/src/lib/utils.ts
Comment (matejchalk) [inline packages/plugin-js-packages/src/lib/utils.ts:1]: Probably `constants.ts` would be a better name than `utils.ts`, based on the content.

File: packages/plugin-js-packages/src/lib/utils.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  npm: 'https://docs.npmjs.com/',
+  'yarn-classic': 'https://classic.yarnpkg.com/docs/',
+  'yarn-modern': 'https://yarnpkg.com/getting-started',
+  pnpm: 'https://pnpm.io/pnpm-cli',
+};
+export const auditDocs: Record<PackageManager, string> = {
+  npm: 'https://docs.npmjs.com/cli/commands/npm-audit',
+  'yarn-classic': 'https://classic.yarnpkg.com/docs/cli/audit',
+  'yarn-modern': 'https://yarnpkg.com/cli/npm/audit',
+  pnpm: 'https://pnpm.io/',
```
Comment (matejchalk) [inline packages/plugin-js-packages/src/lib/utils.ts:28]:
```suggestion
  pnpm: 'https://pnpm.io/cli/audit',
```

File: packages/plugin-js-packages/src/lib/utils.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  'yarn-modern': 'https://github.com/mskelton/yarn-plugin-outdated',
+  pnpm: 'https://pnpm.io/cli/outdated',
+};
+
+export const groupDepDocs: Record<PackageManager, string> = {
+  npm: 'https://docs.npmjs.com/cli/using-npm/config#include',
+  'yarn-classic': 'https://classic.yarnpkg.com/docs/usage',
+  'yarn-modern': 'https://yarnpkg.com/cli/npm/audit#options',
+  pnpm: 'https://pnpm.io/cli/audit#options',
+};
```
Comment (matejchalk) [inline packages/plugin-js-packages/src/lib/utils.ts:43]: These are best official docs on dependency types I could find: https://classic.yarnpkg.com/en/docs/dependency-types

The concept is the same for all packagers. So instead having different links per package manager, I would create different links per dependency type:
- `prod`: https://classic.yarnpkg.com/en/docs/dependency-types#toc-dependencies
- `dev`: https://classic.yarnpkg.com/en/docs/dependency-types#toc-devdependencies
- `optional`: https://classic.yarnpkg.com/en/docs/dependency-types#toc-optionaldependencies

Comment (Tlacenka) [reply packages/plugin-js-packages/src/lib/utils.ts]: good point

**Review by matejchalk — APPROVED** [review]

**Pull request merged**